src/Events/Room/index.js

```diff
@@ -1,14 +1,16 @@
 import { SOCKET_EVENTS } from '../../Constants';
 import RoomService from '../../Services/Room';
+import MessageService from '../../Services/Message';
 
-export const joinRoom = async ({ id, userId }, { socket, io }) => {
+export const joinRoom = async ({ id, userId, username }, { socket, io }) => {
   let room = await RoomService.findRoom(id);
   socket.join(room.id);
   let socketId = socket.id;
-  console.log(`User with ID: ${socketId} joined to room: ${room.id}`);
   room = await RoomService.updateUserId({ id, userId, socketId });
 
-  socket.emit('joined', { room });
+  let message = `${username} joined to room`;
+  await MessageService.saveSystemMessage(id, message);
+  socket.emit(SOCKET_EVENTS.JOINED, { room, message });
```

Comment:
```suggestion
  socket.emit(SOCKET_EVENTS.JOINED, { room });
   //io.to(id).emit(SOCKET_EVENTS.RECEIVE_MESSAGE ...
```
Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.

Reply:
> Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.

Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.

Reply:
> > Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.
> 
> Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.


Nasıl yani? `Chat` içine sistem mesajları için yeni bir prop mu açılacak yoksa şu anda gönderilen mesajların olduğu propsa mı eklenecek. Haklısın api tarafında mesaj olayını da güncellememiz lazım çünkü sadece `text` ve `user` bilgisini gönderiyoruz, mesaj id'si, saat gibi bilgileri de göndrmemiz gerekiyor artık diğer işlemleri yapabilmek için. Mesajı komple emit etmek daha doğru olur bence text, user diye parçalamaktansa

Reply:
> > > Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.
> > 
> > 
> > Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.
> 
> Nasıl yani? `Chat` içine sistem mesajları için yeni bir prop mu açılacak yoksa şu anda gönderilen mesajların olduğu propsa mı eklenecek. Haklısın api tarafında mesaj olayını da güncellememiz lazım çünkü sadece `text` ve `user` bilgisini gönderiyoruz, mesaj id'si, saat gibi bilgileri de göndrmemiz gerekiyor artık diğer işlemleri yapabilmek için. Mesajı komple emit etmek daha doğru olur bence text, user diye parçalamaktansa

Sistem mesajı için yeni bir props gönderilmeli demek istiyorum. Mesajı komple emitlemek mantıklı. Ama fe'de mesajı emitlemek için click fonksiyonunu kullanıyoruz. Yani bir buton yardımıyla mesajı gönderiyoruz. Peki odaya katılma durumunda bunu nasıl yollayacaksın?

Reply:
> > > > Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.
> > > 
> > > 
> > > Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.
> > 
> > 
> > Nasıl yani? `Chat` içine sistem mesajları için yeni bir prop mu açılacak yoksa şu anda gönderilen mesajların olduğu propsa mı eklenecek. Haklısın api tarafında mesaj olayını da güncellememiz lazım çünkü sadece `text` ve `user` bilgisini gönderiyoruz, mesaj id'si, saat gibi bilgileri de göndrmemiz gerekiyor artık diğer işlemleri yapabilmek için. Mesajı komple emit etmek daha doğru olur bence text, user diye parçalamaktansa
> 
> Sistem mesajı için yeni bir props gönderilmeli demek istiyorum. Mesajı komple emitlemek mantıklı. Ama fe'de mesajı emitlemek için click fonksiyonunu kullanıyoruz. Yani bir buton yardımıyla mesajı gönderiyoruz. Peki odaya katılma durumunda bunu nasıl yollayacaksın?

Emitlemekten kastım API --> Frontend yönündeydi, odaya katılma durumunu yeni socket bağlantısı oluştuğu için be takip edebiliyor ve doğrudan oluşturulan sistem mesajını gönderebilir. İki ayrı prop'tan veri alıp bu veriyi tek yerde nasıl yapacağız bilmiyorum. Aklıma ilk gelen şey tarihlerine göre tek bir dizide toplamak olabilir ve ayrıca şu an şunu fark ettim  biz rediste mesajlarla sistem mesajlarını aynı yerde tuttuğumuz ve `room.messages` kısmını doğrudan `Chat` içine gönderdiğimiz için eğer iki prop kullanmak istiyorsak bunları veritabanından tek geldiği için fe'de parçalamamız ve öyle `Chat`'e göndermemiz lazım. Nasıl bir işleyiş yapabiliriz sence?

Reply:
> > > > > Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.
> > > > 
> > > > 
> > > > Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.
> > > 
> > > 
> > > Nasıl yani? `Chat` içine sistem mesajları için yeni bir prop mu açılacak yoksa şu anda gönderilen mesajların olduğu propsa mı eklenecek. Haklısın api tarafında mesaj olayını da güncellememiz lazım çünkü sadece `text` ve `user` bilgisini gönderiyoruz, mesaj id'si, saat gibi bilgileri de göndrmemiz gerekiyor artık diğer işlemleri yapabilmek için. Mesajı komple emit etmek daha doğru olur bence text, user diye parçalamaktansa
> > 
> > 
> > Sistem mesajı için yeni bir props gönderilmeli demek istiyorum. Mesajı komple emitlemek mantıklı. Ama fe'de mesajı emitlemek için click fonksiyonunu kullanıyoruz. Yani bir buton yardımıyla mesajı gönderiyoruz. Peki odaya katılma durumunda bunu nasıl yollayacaksın?
> 
> Emitlemekten kastım API --> Frontend yönündeydi, odaya katılma durumunu yeni socket bağlantısı oluştuğu için be takip edebiliyor ve doğrudan oluşturulan sistem mesajını gönderebilir. İki ayrı prop'tan veri alıp bu veriyi tek yerde nasıl yapacağız bilmiyorum. Aklıma ilk gelen şey tarihlerine göre tek bir dizide toplamak olabilir ve ayrıca şu an şunu fark ettim biz rediste mesajlarla sistem mesajlarını aynı yerde tuttuğumuz ve `room.messages` kısmını doğrudan `Chat` içine gönderdiğimiz için eğer iki prop kullanmak istiyorsak bunları veritabanından tek geldiği için fe'de parçalamamız ve öyle `Chat`'e göndermemiz lazım. Nasıl bir işleyiş yapabiliriz sence?

Zaten `joined` olayında sistem mesajını gönderiyorum. İki ayrı propla hangisinin sistem mesajı hangisinin normal mesaj olduğunu anlayabileceğiz eğer şimdiki şekli kullanırsak. 
Benim aklıma gelen fikir şu: Odaya katıldığında sistem mesajı veritabanına kaydedilsin ve sonra `joined` olayı emitlenerek fe'ye gönderilsin. Fe tarafında da `joined` olayında yakalansın ve `systemMessage` adlı bir değişkende tutulsun ve bu değişken de `chat` komponentine props olarak gönderilsin.

Reply:
> > > > > > Arayüzde kullanıcı odaya katıldığında farklı işlemler yapıyoruz, mesaj gönderildiğinde ise farklı işlemler yapıyoruz. Eğer bu iki işi tek olay içinde gönderirsek fe tarafında da mesaj komponentinin içinde joined olayını da dinlememiz gerekecektir. Bence kullanıcı odaya katıldığında ilk önce "odaya katıldı" mesajını gönderelim ardından ise sistem mesajını normal bir mesaj gönderir gibi gönderelim böylece fe tarafındaki işleyişi de değiştirmemiş olacağız.
> > > > > 
> > > > > 
> > > > > Dinlememizin aksine gelen sistem mesajını `chat` komponentine props olarak göndeririz diye düşündüm. Normal bir mesaj olarak göndermek istesek de yine fe'de değişiklik yapmamız gerekecek. Sonuçta iki tip mesajımız var. İki taraftaki mesaj olayları da normal mesajlara göre yazılmış. Her türlü iki taraftan birinde değişiklik yapılacak.
> > > > 
> > > > 
> > > > Nasıl yani? `Chat` içine sistem mesajları için yeni bir prop mu açılacak yoksa şu anda gönderilen mesajların olduğu propsa mı eklenecek. Haklısın api tarafında mesaj olayını da güncellememiz lazım çünkü sadece `text` ve `user` bilgisini gönderiyoruz, mesaj id'si, saat gibi bilgileri de göndrmemiz gerekiyor artık diğer işlemleri yapabilmek için. Mesajı komple emit etmek daha doğru olur bence text, user diye parçalamaktansa
> > > 
> > > 
> > > Sistem mesajı için yeni bir props gönderilmeli demek istiyorum. Mesajı komple emitlemek mantıklı. Ama fe'de mesajı emitlemek için click fonksiyonunu kullanıyoruz. Yani bir buton yardımıyla mesajı gönderiyoruz. Peki odaya katılma durumunda bunu nasıl yollayacaksın?
> > 
> > 
> > Emitlemekten kastım API --> Frontend yönündeydi, odaya katılma durumunu yeni socket bağlantısı oluştuğu için be takip edebiliyor ve doğrudan oluşturulan sistem mesajını gönderebilir. İki ayrı prop'tan veri alıp bu veriyi tek yerde nasıl yapacağız bilmiyorum. Aklıma ilk gelen şey tarihlerine göre tek bir dizide toplamak olabilir ve ayrıca şu an şunu fark ettim biz rediste mesajlarla sistem mesajlarını aynı yerde tuttuğumuz ve `room.messages` kısmını doğrudan `Chat` içine gönderdiğimiz için eğer iki prop kullanmak istiyorsak bunları veritabanından tek geldiği için fe'de parçalamamız ve öyle `Chat`'e göndermemiz lazım. Nasıl bir işleyiş yapabiliriz sence?
> 
> Zaten `joined` olayında sistem mesajını gönderiyorum. İki ayrı propla hangisinin sistem mesajı hangisinin normal mesaj olduğunu anlayabileceğiz eğer şimdiki şekli kullanırsak. Benim aklıma gelen fikir şu: Odaya katıldığında sistem mesajı veritabanına kaydedilsin ve sonra `joined` olayı emitlenerek fe'ye gönderilsin. Fe tarafında da `joined` olayında yakalansın ve `systemMessage` adlı bir değişkende tutulsun ve bu değişken de `chat` komponentine props olarak gönderilsin.

Peki niye sistem mesajlarını ve normal mesajları ayırıyoruz ve bunları nasıl düzgün sıra ile alt alta gösterebiliriz?